### DISCUSSION TITLE
Converting *.tak to mp3, flac or wav gives me "exit code 224"

### DISCUSSION THREAD

**Discussion by thekryz** [enzo1982/freac#500] [Help & Support]
I got some *.tak files, along with a .log and a .cue file into my hands and would like to convert them into mp3. However, whenever progress seems to be finished, I get an empty export folder and the following error page:
![Bildschirmfoto 2023-06-16 um 22 37 54](https://github.com/enzo1982/freac/assets/10432441/f5e82da6-6952-49c4-a211-257e962025b5)
(for anyone using text search: itcontains "Decoder returned exit code 224" and "Decoder quit prematuerly")

I'm on macOS Ventura 13.4 using fre:ac v1.1.7. I tried converting to FLAC or WAV alternatively, with similar results.

Any ideas? Clicking on the three errors doesn't reveal any additional information. Maybe it has to do with special characters (Japanese) in the artist and title of the music?

**Comment by Lee-Carre** [comment]
> Decoder returned exit code 224

Odd.

A [search of the codebase for “224”](/search?q=repo:enzo1982/freac%20224&type=code) yields nothing which (to a non-programmer) seems to explain what that error code means. Though, there were several matches in a source file which was associated with LAME.

Since its a decoder error, perhaps enquire at somewhere more specific to TAK (perhaps even the developers), what the error code might mean. Fre:AC uses libraries, from elsewhere, to handle the decoding & encoding of various formats, rather than having developed its own implementation. Thus, if the problem is identified & solved upstream, then it'll end up being applied to Fre:AC when enzo1982 updates the dependency.

* Is there anything unusual about your TAK files?
* How were they generated?
  * Else, where did they come from or how did you acquire them (without giving personal details, just the nature of where/how)?
* Does other software have any problems (including throwing errors or warnings in any log of message history) decoding them?
  * What does FFprobe (part of FFmpeg) say, when interrogating them?
* What happens when you try to output something other than MP3s?
* What about a source other than those TAKs, to MP3s?
* Is this error 224 persistent/consistent; occurring every time you try to transcode from the TAK files? Any/all TAK files? What about trying different ones?

I'm wondering if using a cuesheet might have anything to do with it.

In general, try to narrow down the source problem, to isolate it. Then, once it's clearer what needs to be examined, it'll be easier to determine the problem.

**Reply by thekryz** [reply]
I can answer some of your questions more quickly than others:

"Is there anything unusual about your TAK files?"
Not that I know of, but it's the first time I have TAK files at all.

"Where did they come from"
Downloaded.

"What does FFprobe (part of FFmpeg) say, when interrogating them?"
```
(base) me@my-machine ED % ffprobe myfile.tak 
ffprobe version 6.0 Copyright (c) 2007-2023 the FFmpeg developers
  built with Apple clang version 14.0.3 (clang-1403.0.22.14.1)
  configuration: --prefix=/usr/local/Cellar/ffmpeg/6.0_1 --enable-shared --enable-pthreads --enable-version3 --cc=clang --host-cflags= --host-ldflags= --enable-ffplay --enable-gnutls --enable-gpl --enable-libaom --enable-libaribb24 --enable-libbluray --enable-libdav1d --enable-libmp3lame --enable-libopus --enable-librav1e --enable-librist --enable-librubberband --enable-libsnappy --enable-libsrt --enable-libsvtav1 --enable-libtesseract --enable-libtheora --enable-libvidstab --enable-libvmaf --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libxvid --enable-lzma --enable-libfontconfig --enable-libfreetype --enable-frei0r --enable-libass --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libspeex --enable-libsoxr --enable-libzmq --enable-libzimg --disable-libjack --disable-indev=jack --enable-videotoolbox --enable-audiotoolbox
  libavutil      58.  2.100 / 58.  2.100
  libavcodec     60.  3.100 / 60.  3.100
  libavformat    60.  3.100 / 60.  3.100
  libavdevice    60.  1.100 / 60.  1.100
  libavfilter     9.  3.100 /  9.  3.100
  libswscale      7.  1.100 /  7.  1.100
  libswresample   4. 10.100 /  4. 10.100
  libpostproc    57.  1.100 / 57.  1.100
Input #0, tak, from 'myfile.tak':
  Duration: 00:15:31.51, start: 0.000000, bitrate: 894 kb/s
  Stream #0:0: Audio: tak, 44100 Hz, 2 channels, s16p
```

"What happens when you try to output something other than MP3s?"
As stated before, same problem with WAV or FLAC output.

"What about a source other than those TAKs, to MP3s?"
Works fine (tried converting a WAV to MP3)

"Is this error 224 persistent/consistent; occurring every time you try to transcode from the TAK files? Any/all TAK files? What about trying different ones?"
It seems that it might be a problem with the cue sheet. Converting the whole TAK to FLAC without splitting worked. So my workaorund for now is to transcode to FLAC, then split with the CUE & encode to MP3 (Transcoding the whole TAK to MP3 worked as well). Could it be an encoding issue with the CUE? It worked fine to split the FLAC, but I used MediaHuman Audio Converter for that.

**Reply by Lee-Carre** [reply]
> It seems that it might be a problem with the cue sheet. Converting the whole TAK to FLAC without splitting worked. So my workaorund for now is to transcode to FLAC, then split with the CUE & encode to MP3 (Transcoding the whole TAK to MP3 worked as well). Could it be an encoding issue with the CUE? It worked fine to split the FLAC, but I used MediaHuman Audio Converter for that.

Sounds like you're on to something, there.

Unfortunately, I'm not familiar with CueSheets (beyond knowing what they are).

More testing & experimentation.
